File: solutions/python/transpose/1/transpose.py
def transpose(text):
    if not text:
        return ""
    
    lines = text.splitlines()
    max_len = max(len(line) for line in lines)
    
    # Transpose logic with padding handling
    res = []
    for i in range(max_len):
        row = ""
        for j, line in enumerate(lines):
            if i < len(line):
                row += line[i]
            elif any(i < len(l) for l in lines[j+1:]):
                row += " "
        res.append(row)
        
    return "\n".join(res)

File: solutions/ruby/transpose/1/transpose.rb
class Transpose
  def self.transpose(input)
    return "" if input.empty?
    lines = input.split("\n")
    max_length = lines.map(&:length).max
    max_length.times.map do |i|
      chars = lines.map { |l| i < l.length ? l[i] : nil }
      last_real = chars.rindex { |c| !c.nil? }
      chars[0..last_real].map { |c| c.nil? ? " " : c }.join
    end.join("\n")
  end
end
